File: solutions/0896. Monotonic Array/0896.cs
// Time complexity: O(n)
// Space complexity: O(1)
public class Solution {
    public bool IsMonotonic(int[] nums) {
        bool increasing = true;
        bool decreasing = true;

        for (int i = 1; i < nums.Length; ++i) {
            increasing &= nums[i] >= nums[i - 1];
            decreasing &= nums[i] <= nums[i - 1];
        }

        return increasing || decreasing;
    }
}

File: solutions/0896. Monotonic Array/0896.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        increasing = True
        decreasing = True
    
        for i in range(1, len(nums)):
            increasing &= nums[i] >= nums[i - 1]
            decreasing &= nums[i] <= nums[i - 1]

        return increasing or decreasing